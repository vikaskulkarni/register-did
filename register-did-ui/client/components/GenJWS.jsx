import React, { Fragment, Component } from "react";
import "./GenJWS.scss";

const copyToClipBoard = (jws) => {
  var tempInput = document.createElement("input");
  tempInput.value = jws.payload;
  document.body.appendChild(tempInput);
  tempInput.select();
  document.execCommand("copy");
  document.body.removeChild(tempInput);

  console.log("copied");
};

class GenJWS extends Component {
  constructor(props) {
    super(props);
    !props.jwsValue && this.props.resetNextButton(false);
  }

  render() {
    return (
      <Fragment>
        <button
          className="btn btn-info"
          type="button"
          onClick={this.props.genJWS}
        >
          Generate JWT Web Signature
        </button>
        <div className="py-2">
          <div
            data-testid="copy-icon"
            className={this.props.jwsValue ? "copy-icon" : "copy-icon-disabled"}
            onClick={() => {
              copyToClipBoard(this.props.jwsValue);
              this.props.resetNextButton(true);
            }}
          >
            <i className="fa fa-copy">Copy Payload</i>
          </div>
          <textarea
            style={{ resize: "none" }}
            value={JSON.stringify(this.props.jwsValue, 0, 4)}
            rows={10}
            cols={94}
            readOnly
            wrap="hard"
          />
        </div>
      </Fragment>
    );
  }
}

export default GenJWS;
