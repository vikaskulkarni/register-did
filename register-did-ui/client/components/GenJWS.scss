.copy-icon {
    pointer-events: all;
    position: absolute;
    right: 18%;
    cursor: pointer;
    color: #3379b7;
    font-size: 1.1rem;
}

.copy-icon-disabled {
    pointer-events: none;
    position: absolute;
    right: 18%;
    cursor: default;
    color: gray;
    font-size: 1.1rem;
}
