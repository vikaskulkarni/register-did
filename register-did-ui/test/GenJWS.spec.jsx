import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';

import GenJWS from '../client/components/GenJWS.jsx'

test('renders GenJWS as expected', () => {
   const resetHandler = (value) => {
      expect(value).toBe(true);
   }
   const tree = renderer.create(<GenJWS jwsValue="dummy data" resetNextButton={resetHandler} />).toJSON();
   expect(tree).toMatchSnapshot();
});

test('when no jwt value, does not allow copy and rest', () => {
   const resetHandler = jest.fn();
   const {getByTestId} = render(<GenJWS resetNextButton={resetHandler} />);
   expect(getByTestId('copy-icon')).toHaveAttribute('class', 'copy-icon-disabled');
});

test('clicking on copy icon calls reset handler once', () => {
   document.execCommand = jest.fn();
   const resetHandler = jest.fn();
   const {getByTestId} = render(<GenJWS jwsValue="dummy data" resetNextButton={resetHandler} />);
   fireEvent.click(getByTestId('copy-icon'));
   expect(resetHandler).toBeCalledTimes(1);
});
