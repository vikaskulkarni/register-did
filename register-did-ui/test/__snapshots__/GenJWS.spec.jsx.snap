// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GenJWS as expected 1`] = `
Array [
  <button
    className="btn btn-info"
    type="button"
  >
    Generate JWT Web Signature
  </button>,
  <div
    className="py-2"
  >
    <div
      className="copy-icon"
      data-testid="copy-icon"
      onClick={[Function]}
    >
      <i
        className="fa fa-copy"
      >
        Copy Payload
      </i>
    </div>
    <textarea
      cols={94}
      readOnly={true}
      rows={10}
      style={
        Object {
          "resize": "none",
        }
      }
      value="\\"dummy data\\""
      wrap="hard"
    />
  </div>,
]
`;
